tracardi/process_engine/action/v1/password_generator_action.py

```diff
@@ -0,0 +1,142 @@
+from pydantic import validator
+
+from tracardi.service.plugin.runner import ActionRunner
+from tracardi.service.plugin.domain.result import Result
+from tracardi.service.plugin.domain.register import Plugin, Spec, MetaData, Documentation, PortDoc, Form, FormGroup, \
+    FormComponent, FormField
+from tracardi.service.plugin.domain.config import PluginConfig
+
+from password_generator import PasswordGenerator
+
+
+class Config(PluginConfig):
+    max_length: int
+    min_length: int
+    uppercase: int
+    lowercase: int
+    special_characters: int
+
+    @validator("min_length")
+    def check_min_max_value(cls, value, values):
+        if value > values["max_length"]:
+            raise ValueError("Minimum length is bigger than maximum length")
+        return value
+
+
+def validate(config: dict) -> Config:
+    return Config(**config)
+
+
+class PasswordGeneratorAction(ActionRunner):
+    def __init__(self, **kwargs):
+        self.pgo = PasswordGenerator()
+        self.config = validate(kwargs)
+        self.pgo.minlen = self.config.min_length
+        self.pgo.maxlen = self.config.max_length
+        self.pgo.minuchars = self.config.uppercase
+        self.pgo.minlchars = self.config.lowercase
+        self.pgo.minschars = self.config.special_characters
+
+    async def run(self, payload: dict, in_edge=None) -> Result:
+        password = self.pgo.generate()
+        return Result(port="password", value={"password": password})
+
+
+def register() -> Plugin:
+    return Plugin(
+        start=False,
+        spec=Spec(
+            module=__name__,
+            className='PasswordGeneratorAction',
+            inputs=["payload"],
+            outputs=["password"],
+            version='0.7.2',
+            license="MIT",
+            author="Mateusz Zitaruk",
+            init={"min_length": None,
```

Comment:
I would set default values for these properties. This way users can quickly set up the plugin. The default values are up to you.